### ISSUE TITLE
Logstash 2.2 breakage

### ISSUE THREAD

**Issue by lothor** [magnusbaeck/logstash-filter-verifier#4]
Hi,
Having recently upgraded to logstash 2.2, logstash-filter-verifier has stopped working.

This is caused by the --filterworkers-parameter which seems to have been removed / replaced as a result of the pipeline changes in logstash 2.2:

    $ ./logstash-filter-verifier --loglevel=DEBUG example.json logstash.config
    2016/02/10 19:13:55 Reading test case file: example.json (/go/src/github.com/magnusbaeck/logstash-filter-verifier/example.json)
    2016/02/10 19:13:55 Starting "/opt/logstash/bin/logstash" with args ["/opt/logstash/bin/logstash" "--filterworkers" "1" "--debug" "-e" "input { stdin { codec => \"plain\" add_field => { \"type\" => \"syslog\" } } } output { file { path => \"/tmp/168470069\" codec => \"json_lines\" } }" "--log" "/tmp/780283920" "--config" "logstash.config"].
    2016/02/10 19:13:55 Waiting for child with pid 9549 to terminate.
    exit status 1

Since the error message from logstash isn't currently available, the actual error can be reproduced by running logstash manually using the command used by the verifier:

    $ "/opt/logstash/bin/logstash" "--filterworkers" "1" "--debug" "-e" "input { stdin { codec => \"plain\" add_field => { \"type\" => \"syslog\" } } } output { file { path => \"/tmp/168470069\" codec => \"json_lines\" } }" "--log" "/tmp/780283920" "--config" "logstash.config"
    Clamp::UsageError: Unrecognised option '--filterworkers'
      signal_usage_error at /opt/logstash/vendor/bundle/jruby/1.9/gems/clamp-0.6.5/lib/clamp/command.rb:103
             find_option at /opt/logstash/vendor/bundle/jruby/1.9/gems/clamp-0.6.5/lib/clamp/option/parsing.rb:62
           parse_options at /opt/logstash/vendor/bundle/jruby/1.9/gems/clamp-0.6.5/lib/clamp/option/parsing.rb:28
                   parse at /opt/logstash/vendor/bundle/jruby/1.9/gems/clamp-0.6.5/lib/clamp/command.rb:52
                     run at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-2.2.0-java/lib/logstash/runner.rb:79
                    call at org/jruby/RubyProc.java:281
                     run at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-2.2.0-java/lib/logstash/runner.rb:95
                    call at org/jruby/RubyProc.java:281
              initialize at /opt/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.22/lib/stud/task.rb:24

(Thanks for the excellent tool btw, love the simplicity of using it compared to our old rspec-tests for our filters)

**Comment by magnusbaeck** [comment]
Thanks for debugging this! I got a similar report earlier today (issue #3) and it's probably the same root cause (although a syntax error in the config file would produce the same terrible error message).

So Logstash 2.2 removed `--filterworkers`, just like that. I've filed https://github.com/elastic/logstash/issues/4669 so that at least the documentation can be corrected. Luckily the `-w` shortcut is still there so I can use that instead and get something that works in both 2.1 and 2.2. I've pushed commit https://github.com/magnusbaeck/logstash-filter-verifier/commit/d4094a0e5993cb726d7db44905efc92d0363b5d9 to that effect and will follow up later this week with a 1.0.1 release that fixes the error message bug.
